holidays/countries/tuvalu.py

```diff
@@ -0,0 +1,201 @@
+    """
+
+    country = "TV"
+    default_language = "tvl_TV"
+    # %s (estimated).
+    estimated_label = tr("%s (fakaaoga)")
+    # %s (observed).
+    observed_label = tr("%s (fakamatakuga)")
+    # %s (observed, estimated)
+    observed_estimated_label = "%s (fakamatakuga, fakaaoga)"
+    supported_languages = ("en_GB", "en_US", "tvl_TV")
+    subdivisions = (
+        "FUN",  # Funafuti.
+        "NMG",  # Nanumaga.
+        "NMA",  # Nanumea.
+        "NIT",  # Niutao.
+        "NUI",  # Nui.
+        "NKF",  # Nukufetau.
+        "NKL",  # Nukulaelae.
+        "VAI",  # Vaitupu.
+    )
+    subdivisions_aliases = {
+        # Town/Island Councils.
+        "Funafuti": "FUN",
+        "Nanumaga": "NMG",
+        "Nanumea": "NMA",
+        "Niutao": "NIT",
+        "Nui": "NUI",
+        "Nukufetau": "NKF",
+        "Nukulaelae": "NKL",
+        "Vaitupu": "VAI",
+    }
+    # Tuvalu became fully independent of the United Kingdom on October 1, 1978
+    # Tuvalu's PUBLIC HOLIDAYS (AMENDMENT) ACT 1990 (Act 2 of 1990)
+    # It was first proclaimed on FEB 7th, 1990
+    start_year = 1990
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # Public Holidays
+
+        # New Year's Day.
+        self._add_new_years_day(tr("Tausaga Fou"))
+
+        # Commonwealth Day.
+        self._add_holiday_2nd_mon_of_mar(tr("Aso Atefenua"))
+
+        # Gospel Day.
+        self._add_holiday_2nd_mon_of_may(tr("Te Aso o te Tala Lei"))
+
+        # King's Birthday.
+        self._add_holiday_2nd_sat_of_jun(tr("Asofanau Tupu"))
+
+        # National Children's Day.
+        name = tr("Aso Tamaliki")
+
+        if self._year >= 2019:
+            # National Children's Day.
+            self._add_holiday_1_day_past_2nd_sun_of_oct(name)
+
+            # National Youth Day.
+            self._add_holiday_1st_mon_of_aug(tr("Aso tupulaga"))
+        else:
+            # National Children's Day.
+            self._add_holiday_1st_mon_of_aug(name)
+
+        # Tuvalu Day.
+        self._add_holiday_oct_1(tr("Tutokotasi"))
+
+        # Tuvalu Day.
+        self._add_holiday_oct_2(tr("Tutokotasi"))
+
+        # Referenced : https://en.wikipedia.org/wiki/King%27s_Official_Birthday#Tuvalu
+        if self._year >= 2021:
+            # Heir to the Throne's Birthday.
+            self._add_holiday_2nd_mon_of_nov(tr("Aso fanau o te sui ote Tupu"))
+
+        # Boxing Day.
+        self._add_christmas_day_two(tr("Aso Faipele"))
+
+        # Religious Holidays
+
+        # Good Friday.
+        self._add_good_friday(tr("Aso toe tu"))
+
+        # Holy Saturday.
+        self._add_observed(self._add_holy_saturday(tr("Aso Tapu")))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Toe Tu aso gafua"))
```

Comment:
It's more reasonable to place these holidays in chronological order among others, rather than separately.